### PR TITLE
Support UVs on initial shapes

### PR DESCRIPTION
**Pull request by bneukom**

### CODE REVIEW

File: VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp [hunk 1]
```diff
@@ -383,6 +376,34 @@ TArray<TArray<FSplinePoint>> CreateSplinePointsFromInitialFaces(const TArray<FIn
...
+
+	const FVector PlaneNormal(PlaneNormalOut.X, PlaneNormalOut.Y,PlaneNormalOut.Z);
+	float Dot = FVector::DotProduct(FVector::UpVector, PlaneNormal);
+	
+	if (Dot < 0)
+	{
+		Algo::Reverse(Vertices);
+		Algo::Reverse(TextureCoordinates);
+	}
+}
```
Comment (chr11115) [inline VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp:405]: Nice hidden cleanup ;)

**Review by chr11115 — CHANGES_REQUESTED** [review]
Nice job, However there seems to be an issue with indexing. Opening the default scene throws a range error in the FTextureCoordinateSet:
image


Also a few readability issues/cleanup suggestions.

File: VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioModule.cpp [hunk 2]
```diff
@@ -148,6 +148,26 @@ void SetInitialShapeGeometry(const InitialShapeBuilderUPtr& InitialShapeBuilder,
 	{
 		UE_LOG(LogUnrealPrt, Error, TEXT("InitialShapeBuilder setGeometry failed status = %hs"), prt::getStatusDescription(SetGeometryStatus))
 	}
+	
+	for (int32 UVSet = 0; UVSet < 8; ++UVSet)
+	{
+		std::vector<double> uvCoords;
+		for (const FInitialShapeFace& Face : InitialShape)
+        {
+            for (const auto& UV : Face.TextureCoordinates[UVSet].TextureCoordinates)
```
Comment (chr11115) [inline VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioModule.cpp:157]: make sure the TextureCoordinates are initialized, before accessing it, see Error, when opening default scene:
image

File: VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp (hunk 1 above)
Comment (bneukom) [reply VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp:405]: Hehe ;) Should maybe have gone into a separate commit!

File: VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioModule.cpp (hunk 2 above)
Comment (bneukom) [reply VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioModule.cpp:157]: Good catch! Will fix that!

**Comment by chr11115** [comment]
LGTM, Did quick functional test (see internal ticket)

**Review by chr11115 — APPROVED** [review]

**Pull request merged**